### PR TITLE
Expose free text equality and diversity data in Vendor API

### PR DESCRIPTION
**Pull request by benswannack**
## Context
We have been asked to expose this data since Vendors will need more information than just "Other" if we have it.

## Link to Trello card
https://trello.com/c/lVL9xKPU/3742-tribal-selected-disability-other-and-populated-textbox-the-text-is-not-sent-in-the-api

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/presenters/vendor_api/single_application_presenter.rb [hunk 1]
```diff
@@ -369,8 +369,26 @@ def hesa_itt_data
...
-        }
+        }.merge(additional_hesa_itt_data(equality_and_diversity_data))
+      end
+    end
+
+    def additional_hesa_itt_data(equality_and_diversity_data)
+      standard_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map(&:last)
+      other_disability_text = (equality_and_diversity_data['disabilities'] - standard_disabilities).first.presence
+
+      other_ethnicity_code = Hesa::Ethnicity.find('Another ethnic background', application_form.recruitment_cycle_year)&.hesa_code
```
Comment (benswannack) [inline app/presenters/vendor_api/single_application_presenter.rb:380]: There are other ways of saying a candidate has "Other" ethnicity, but they are a bit more specific like "Other white background" & they have different codes, so I've left them alone here

File: config/vendor-api-v1.yml [hunk 2]
```diff
@@ -1175,6 +1175,16 @@ components:
...
+        other_disability_details:
+          type: string
+          nullable: true
+          description: The candidate’s description of their disability, if they selected “Other”. Corresponds to the HESA disability code `96`
+          example: "Partially sighted"
+        other_ethnicity_details:
+          type: string
+          nullable: true
+          description: The candidate’s description of their ethnicity, if they selected “Other”. Corresponds to the HESA ethnicity code `80`
+          example: "Hawaiian"
```
Comment (benswannack) [inline config/vendor-api-v1.yml:1187]: I've guessed at these examples with what I think is sensible. Do we have an examples of the intended use of this field?

File: app/presenters/vendor_api/single_application_presenter.rb (hunk 1 above)
Comment (cwrw) [reply app/presenters/vendor_api/single_application_presenter.rb:380]: I think we need to account for any text entered by the user, which includes any "other ..." field

Comment (cwrw) [reply app/presenters/vendor_api/single_application_presenter.rb:380]: We also need to ignore the standard text which could look like "Another Mixed background" which is not entered by the user

File: config/vendor-api-v1.yml (hunk 2 above)
Comment (cwrw) [reply config/vendor-api-v1.yml:1187]: Maybe we keep it ambiguous and have something like "My disability is ..." or something similar, same with ethnicity

File: app/views/api_docs/vendor_api_docs/pages/release_notes.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
+## 3rd June
+
+`HESAITTData` now includes the following optional fields:
+
+- `other_disability_details` will contain the candidate’s description of their disability if they selected “Other”. This corresponds to HESA disability code `96`
+- `other_ethnicity_details` will contain the candidate’s description of their ethnicity if they selected “Other”. This corresponds to HESA ethnicity code `80`
```
Comment (cwrw) [inline app/views/api_docs/vendor_api_docs/pages/release_notes.md:6]: just to double check `80` covers all values under `OTHER_ETHNIC_BACKGROUNDS`?

File: app/presenters/vendor_api/single_application_presenter.rb [hunk 4]
```diff
@@ -369,10 +369,31 @@ def hesa_itt_data
...
+
+    def other_disability_details(equality_and_diversity_data)
+      return unless equality_and_diversity_data['hesa_disabilities'].include?('96')
+
+      standard_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map(&:last)
+      (equality_and_diversity_data['disabilities'] - standard_disabilities).first.presence
+    end
+
+    def other_ethnicity_details(equality_and_diversity_data)
+      known_ethnic_backgrounds = OTHER_ETHNIC_BACKGROUNDS.values + ETHNIC_BACKGROUNDS.values.flatten
```
Comment (cwrw) [inline app/presenters/vendor_api/single_application_presenter.rb:391]: does this cover `Prefer not to say`?

**Review by cwrw — COMMENTED** [review]
looks great, nice refactor. Just some final comments

File: app/views/api_docs/vendor_api_docs/pages/release_notes.md (hunk 3 above)
Comment (benswannack) [reply app/views/api_docs/vendor_api_docs/pages/release_notes.md:6]: Ah good catch, this is no longer the case. I'll remove the mention of the code from here

**Review by cwrw — APPROVED** [review]
⭐

**Pull request merged**